app/SignInWithGithub/SignInWithGithub.tsx

```diff
@@ -0,0 +1,99 @@
+"use client"
+import { signInWithPopup,GithubAuthProvider,signOut } from 'firebase/auth'
+import React, { useEffect, useState } from 'react'
+import {auth,provider} from '../../lib/firebase-config'
+import { useRouter } from 'next/router'
+import Image from 'next/image'
+import { toast,ToastContainer } from 'react-toastify'
+import 'react-toastify/dist/ReactToastify.css';
+
+const SignInWithGithub=()=>{
+    const router = useRouter();
+    const [imageURL, setImageURL] = useState<string | null>(null);
+    const [authenticated,setAuthenticated] = useState(false);
+    const signIn = async () => {
+        try {
+            const result = await signInWithPopup(auth, provider);
+            const credential = GithubAuthProvider.credentialFromResult(result);
+            console.log("Credential",credential);
+            console.log("Result: ",result);
+            const token: any = result.user.accessToken;
```

Comment:
@Anmol-Baranwal, according to my terminal, "Property `accessToken' does not exist on type 'User'", so maybe this needs some changing?